core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -14,6 +14,13 @@ enum class CredentialType {
     SsoProfile
 }
 
+enum class CredentialSourceId {
```

Comment:
Since this is only for telemetry, do we actually need this?

Reply:
We define credentials factory in Core, but we generate telemetry in JB Core, so we either:
1. go unsafe and use strings
2. Dupe the enum for some semblance of type safety.

I voted for 2.